api/renterhost_test.go

```diff
@@ -118,6 +229,38 @@ func TestIntegrationHostAndRent(t *testing.T) {
 		t.Fatal("data mismatch when downloading a file")
 	}
 
+	// Try downloading a nonexistent file.
+	err = st.stdGetAPI("/renter/download/fake?destination=" + downpath)
+	if err.Error() != "Download failed: no file with that path" {
+		t.Errorf("expected error to be 'Download failed: no file with that path'; got %v instead", err)
+	}
```

Comment:
I slammed all of the edge case testing into a single test in the host, and I now regret it because I find it very difficult to look at the test and know what's going on. And if I want to change it, I have to worry about what edge cases I might be changing. (if there's an empty folder test further down I have to move it, for example, but it's also possible to silently make the test useless by setting up conditions where the test will now accidentally always pass).